contrib/node/src/python/pants/contrib/node/subsystems/node_distribution.py

```diff
@@ -10,64 +10,62 @@
 import shutil
 from collections import namedtuple
 
+from pants.base.deprecated import deprecated_conditional
 from pants.base.exceptions import TaskError
 from pants.base.hash_utils import hash_file
+from pants.binaries.binary_tool import NativeTool
 from pants.binaries.binary_util import BinaryUtil
-from pants.fs.archive import TGZ
 from pants.option.custom_types import dir_option, file_option
-from pants.subsystem.subsystem import Subsystem
 from pants.util.dirutil import safe_mkdir, safe_rmtree
-from pants.util.memo import memoized_method
+from pants.util.memo import memoized_method, memoized_property
 from pants.util.process_handler import subprocess
 
+from pants.contrib.node.subsystems.yarnpkg_distribution import YarnpkgDistribution
+
 
 logger = logging.getLogger(__name__)
 
 
-class NodeDistribution(object):
+class NodeDistribution(NativeTool):
```

Comment:
We should preserve `TGZ.extract(tarball_filepath, work_dir, concurrency_safe=True)` functionality that move the extracted tar ball atomically.  Refer to https://github.com/pantsbuild/pants/pull/5248

Reply:
The base class takes care of this.

Reply:
Or at least it was supposed to, let me check.

Reply:
Hmm, looks like it didn't. Good catch! Fixed.